processor/batchprocessor/factory.go

```diff
@@ -44,8 +44,9 @@ func NewFactory() processor.Factory {
 
 func createDefaultConfig() component.Config {
 	return &Config{
-		SendBatchSize: defaultSendBatchSize,
-		Timeout:       defaultTimeout,
+		SendBatchSize:            defaultSendBatchSize,
+		Timeout:                  defaultTimeout,
+		MetadataCardinalityLimit: 100,
```

Comment:
nit: define this in the const above
```suggestion
		MetadataCardinalityLimit: defaultMetadataCardinalityLimit,
```